File: c/10lug20/10lug20.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <math.h>

void memorizza_in_array(int, int*);
int verifica_numero(int, int, int*);

int main(void) {
	int seed, coeff[2], n = 1, i=0;
	seed = time(NULL);
	srand48(seed);
	do {
		n = lrand48()%99+1;
		memorizza_in_array(n,coeff);
		i++;
	} while(verifica_numero(n,i,coeff));
	return 0;
}

void memorizza_in_array(int n, int* coeff) {
	int dec, unit;
	dec = n / 10;
	unit = n%10;
	*coeff = dec;
	*(coeff+1) = unit;
}

int verifica_numero(int n, int i, int* coeff) {
	int boolean = 1, sum,rev;
	sum = coeff[0]+coeff[1];
	printf("n = %d, c1c2 = %d%d, S = %d\n",n,coeff[0],coeff[1],sum);
	rev = coeff[1]*10+coeff[0];
	if (n+sum == rev) {
		printf("Intero che soddisfa la condizione n = %d\nNumero di tentativi: %d\n", n,i);
		boolean = 0;
	}
	return boolean;
}

File: go/10lug20/10lug20.go
package main

import (
	"fmt"
	"math/rand"
	"time"
);

func memorizza_in_array(n int, array [2]int) {
	var dec int;
	var unit int;
	dec = n/10;
	unit = n%10;
	array[0] = dec;
	array[1] = unit;
}

func verifica_numero(n int, array [2]int) bool {
	var exit bool = false;
	var sum, rev int;
	sum = array[0]+array[1];
	fmt.Printf("n = %d, c1c2 = %d%d, S =%d\n",n,array[0],array[1],sum);
	rev = array[1]*10+array[0];
	if (n+sum == rev) {
		fmt.Printf("Intero che soddisfa la condizione n = %d\n", n);
		exit = true;
	}
	return exit;
}

func main() {
	var n int;
	coeff := [2]int{0};
	rand.Seed(time.Now().UnixNano());
	for {
		n = rand.Intn(100);
		memorizza_in_array(n,coeff);
		if (verifica_numero(n,coeff)) {
			break;
		}
	}
	return;
}
